src/jit/lower.cpp

```diff
@@ -108,20 +108,39 @@ bool Lowering::IsSafeToContainMem(GenTree* parentNode, GenTree* childNode)
 // IsContainableMemoryOp: Checks whether this is a memory op that can be contained.
 //
 // Arguments:
-//    node - the node of interest.
+//    node        - the node of interest.
+//    useTracked  - true if this is being called after liveness so lvTracked is correct
+//
+// Return value:
+//    True if this will definitely be a memory reference that could be contained.
 //
 // Notes:
 //    This differs from the isMemoryOp() method on GenTree because it checks for
-//    the case of an untracked local. Note that this won't include locals that
-//    for some reason do not become register candidates, nor those that get
+//    the case of doNotEnregister local. This won't include locals that
+//    for some other reason do not become register candidates, nor those that get
 //    spilled.
+//    Also, if we call this before we redo liveness analysis, any new lclVars
+//    introduced after the last dataflow analysis will not yet be marked lvTracked,
+//    so we don't use that.
 //
-// Return value:
-//    True if this will definitely be a memory reference that could be contained.
-//
-bool Lowering::IsContainableMemoryOp(GenTree* node)
+bool Lowering::IsContainableMemoryOp(GenTree* node, bool useTracked)
```

Comment:
I find it easier to read without the template syntax, and I'm reasonably confident that the compiler will "do the right thing". That said, I should probably have mentioned that this is likely to be temporary, since eventually all the containment analysis will be moving to `Lowering` where we can't rely on `lvTracked`. The mitigation is to make things contained in `TreeNodeInfoInit` if they are lclVars marked `RegOptional`. The only fly in the ointment is that the semantics are not identical (as I mentioned before), since for the latter we don't always check `IsSafeToContainMem` even though I think we should. However that's a topic/change for another PR.